📖 Add MacOS instructions for the CAPD quickstart

erwinvaneyk opened the pull request:
**What this PR does / why we need it**:
Right now, without any modifications, the CAPD instructions do not work on MacOS. This PR adds additional instructions to the quickstart guide on how to get it working.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1534

detiber commented:
/ok-to-test
lgtm at first glance @chuckha

wfernandes commented:
I just tested the sed commands and everything seems to work for me 😄

dlipovetsky commented:
Thanks @erwinvaneyk!

@vincepri I assume we'll cherry-pick this to `master` once #1549 merges

dlipovetsky commented on docs/book/src/tasks/installation.md:
```diff
@@ -75,7 +67,7 @@ EOF
 <h1>Warning</h1>
 
-[kind] is not designed for production use, and is intended for development environments only.
+[kind] is not designed for production use; it is intended for development and testing environments.
```
nit: Make this change in the two other places this warning is used.

dlipovetsky commented on docs/book/src/user/quick-start.md:
````diff
@@ -318,13 +318,44 @@ spec:
 
+{{#tabs name:"tab-getting-kubeconfig" tabs:"AWS,Docker,vSphere"}}
+{{#tab AWS}}
+```bash
````
Add a blank line between L322 and L323

dlipovetsky commented on docs/book/src/user/quick-start.md:
````diff
@@ -318,13 +318,44 @@ spec:
+  > ./capi-quickstart.kubeconfig
+```
+{{#/tab }}
+{{#tab Docker}}
````
```suggestion
{{#tab Docker}}

```

dlipovetsky commented on docs/book/src/user/quick-start.md:
````diff
@@ -318,13 +318,44 @@ spec:
+```  
+{{#/tab }}
+{{#tab vSphere}}
+```bash
````
```suggestion

```bash
```

dlipovetsky commented:
I see a the code block incorrectly rendered in the [netlify preview](https://deploy-preview-1562--kubernetes-sigs-cluster-api.netlify.com/user/quick-start.html): 
![image](https://user-images.githubusercontent.com/3445370/66942739-90b4a400-effe-11e9-99cf-88cb5afdd5b6.png)

I remember having this issue before: mdbook expects a blank line following a `{#tab }` directive. Added suggestions.

nader-ziada commented:
tested the `mac os` instructions and can confirm it works for me, Thanks

chuckha commented:
Once feedback is addressed this LGTM

erwinvaneyk commented:
@dlipovetsky - thanks for the review!

FYI, I accidentally rebased onto master instead of release-0.2 when making the feedback fixes. Fixed it, but it caused a bit of a blurp with the CI. 

Addressed all feedback - should be ready to merge. 🙂

detiber commented:
/approve

chuckha commented:
/lgtm

All feedback was addressed 👍 thanks so much for the instructions!
